src/main/kotlin/io/emeraldpay/dshackle/upstream/CurrentMultistreamHolder.kt

```diff
@@ -139,23 +60,8 @@ open class CurrentMultistreamHolder(
         )
     }
 
-    override fun getDefaultMethods(chain: Chain): CallMethods {
-        return callTargets[chain] ?: return setupDefaultMethods(chain)
-    }
-
-    fun setupDefaultMethods(chain: Chain): CallMethods {
-        val created = when (BlockchainType.from(chain)) {
-            BlockchainType.EVM_POW -> DefaultEthereumMethods(chain)
-            BlockchainType.BITCOIN -> DefaultBitcoinMethods()
-            BlockchainType.EVM_POS -> DefaultEthereumMethods(chain)
-            else -> throw IllegalStateException("Unsupported chain: $chain")
-        }
-        callTargets[chain] = created
-        return created
-    }
-
     override fun isAvailable(chain: Chain): Boolean {
-        return chainMapping.containsKey(chain) && callTargets.containsKey(chain)
+        return chainMapping.getValue(chain).isAvailable()
```

Comment:
why not checked against targets

Reply:
First of all - it doesn't exists here anymore.
Second - it works like cache for call targets and doesn't contains any new information about chains state - do we really need it?